server/src/repositories/library.repository.ts

```diff
@@ -1,84 +1,122 @@
-import { Repository } from 'typeorm/repository/Repository.js';
+
+const userColumns = [
+  'users.id',
+  'users.email',
+  'users.createdAt',
+  'users.profileImagePath',
+  'users.isAdmin',
+  'users.shouldChangePassword',
+  'users.deletedAt',
+  'users.oauthId',
+  'users.updatedAt',
+  'users.storageLabel',
+  'users.name',
+  'users.quotaSizeInBytes',
+  'users.quotaUsageInBytes',
+  'users.status',
+  'users.profileChangedAt',
+] as const;
+
+const withOwner = (eb: ExpressionBuilder<DB, 'libraries'>) => {
+  return jsonObjectFrom(eb.selectFrom('users').whereRef('users.id', '=', 'libraries.ownerId').select(userColumns)).as(
+    'owner',
+  );
+};
 
 @Injectable()
 export class LibraryRepository implements ILibraryRepository {
-  constructor(@InjectRepository(LibraryEntity) private repository: Repository<LibraryEntity>) {}
+  constructor(@InjectKysely() private db: Kysely<DB>) {}
 
   @GenerateSql({ params: [DummyValue.UUID] })
-  get(id: string, withDeleted = false): Promise<LibraryEntity | null> {
-    return this.repository.findOneOrFail({
-      where: {
-        id,
-      },
-      relations: { owner: true },
-      withDeleted,
-    });
+  get(id: string, withDeleted = false): Promise<LibraryEntity | undefined> {
+    return this.db
+      .selectFrom('libraries')
+      .selectAll('libraries')
+      .select(withOwner)
+      .where('libraries.id', '=', id)
+      .$if(!withDeleted, (qb) => qb.where('libraries.deletedAt', 'is', null))
+      .executeTakeFirst() as Promise<LibraryEntity | undefined>;
   }
 
   @GenerateSql({ params: [] })
   getAll(withDeleted = false): Promise<LibraryEntity[]> {
-    return this.repository.find({
-      relations: {
-        owner: true,
-      },
-      order: {
-        createdAt: 'ASC',
-      },
-      withDeleted,
-    });
+    return this.db
+      .selectFrom('libraries')
+      .selectAll('libraries')
+      .select(withOwner)
+      .orderBy('createdAt', 'asc')
+      .$if(!withDeleted, (qb) => qb.where('libraries.deletedAt', 'is', null))
+      .execute() as unknown as Promise<LibraryEntity[]>;
   }
 
   @GenerateSql()
   getAllDeleted(): Promise<LibraryEntity[]> {
-    return this.repository.find({
-      where: {
-        deletedAt: Not(IsNull()),
-      },
-      relations: {
-        owner: true,
-      },
-      order: {
-        createdAt: 'ASC',
-      },
-      withDeleted: true,
-    });
+    return this.db
+      .selectFrom('libraries')
+      .selectAll('libraries')
+      .select(withOwner)
+      .where('libraries.deletedAt', 'is not', null)
+      .orderBy('createdAt', 'asc')
+      .execute() as unknown as Promise<LibraryEntity[]>;
   }
 
-  create(library: Omit<LibraryEntity, 'id' | 'createdAt' | 'updatedAt' | 'ownerId'>): Promise<LibraryEntity> {
-    return this.repository.save(library);
+  create(library: Insertable<Libraries>): Promise<LibraryEntity> {
+    return this.db
+      .insertInto('libraries')
+      .values(library as LibraryEntity)
```

Comment:
Why the cast here?

Reply:
Left over from an earlier method signature, ty!